lib/kubernetes-deploy/cluster_resource_discovery.rb

```diff
@@ -16,8 +16,26 @@ def crds
       end
     end
 
+    def global_resource_names
+      @globals ||= fetch_globals.map { |g| g["kind"] }
+    end
+
     private
 
+    def fetch_globals
+      raw, _, st = kubectl.run("api-resources", "--namespaced=false", output: "wide", attempts: 5)
+      if st.success?
+        rows = raw.split("\n")
+        header = rows.shift.downcase.scan(/[a-z]+[\W]*/).each_with_object({}) do |match, hash|
+          start = hash.values.map(&:last).max.to_i
```

Comment:
This seems to be reflecting on the results of the block up to this point in the iteration. Could we be storing this value as we iterate instead of computing it like this? Isn't it the length of the header string we've processed so far, aka the end of the match we last processed?